Spanner: 'test_reload_instance' systest flakes.

From [this Kokoto run](https://source.cloud.google.com/results/invocations/9c28a3d8-da69-4bca-9b82-db7166c03522/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fspanner/log)

```python
self = <tests.system.test_system.TestInstanceAdminAPI testMethod=test_reload_instance>
    def test_reload_instance(self):
        # Use same arguments as Config.INSTANCE (created in `setUpModule`)
        # so we can use reload() on a fresh instance.
        instance = Config.CLIENT.instance(INSTANCE_ID)
        # Make sure metadata unset before reloading.
        instance.display_name = None
        instance.reload()
>       self.assertEqual(instance.display_name, Config.INSTANCE.display_name)
E       AssertionError: 'GCP System Tests' != 'Foo Bar Baz'
E       - GCP System Tests
E + Foo Bar Baz
```

Maybe somebody tweaked the title in the console while the test was running?  Because it is "Foo Bar Baz" now.

--- a/spanner/tests/system/test_system.py
+++ b/spanner/tests/system/test_system.py
@@ -146,7 +146,13 @@ class TestInstanceAdminAPI(unittest.TestCase):
         # Make sure metadata unset before reloading.
         instance.display_name = None
 
-        instance.reload()
+        def _expected_display_name(instance):
+            return instance.display_name == Config.INSTANCE.display_name
+
+        retry = RetryInstanceState(_expected_display_name)
+
+        retry(instance.reload)()
+
         self.assertEqual(instance.display_name, Config.INSTANCE.display_name)
 
     @unittest.skipUnless(CREATE_INSTANCE, "Skipping instance creation")

